library/src/scala/collection/SeqView.scala

```diff
@@ -151,7 +151,7 @@ object SeqView {
 
       def apply(i: Int): A = _reversed.apply(i)
       def length: Int = len
-      def iterator: Iterator[A] = Iterator.empty ++ _reversed.iterator // very lazy
+      def iterator: Iterator[A]^{this} = Iterator.empty ++ _reversed.iterator // very lazy
```

Comment:
For future head-scratchers: `++` on iterators are by-name on the RHS, and so will capture `this` by virtue of accessing the lazy val `_reversed`